src/EditorFeatures/Core.Wpf/InlineDiagnostics/InlineDiagnosticsAdornmentManager.cs

```diff
@@ -188,7 +188,16 @@ protected override void AddAdornmentsToAdornmentLayer_CallOnlyOnUIThread(Normali
 
         protected override void RemoveAdornmentFromAdornmentLayer_CallOnlyOnUIThread(SnapshotSpan span)
         {
-            AdornmentLayer.RemoveAdornmentsByVisualSpan(new SnapshotSpan(span.Start.GetContainingLine().Start, span.End.GetContainingLine().End));
+            var lineSpan = new SnapshotSpan(span.Start.GetContainingLine().Start, span.End.GetContainingLine().End);
+            AdornmentLayer.RemoveMatchingAdornments(e =>
```

Comment:
You may want to add a comment here why we're not calling RemoveAdornmentsByVisualSpan.